### PR TITLE
Add WebPlugin to keep running in background

### PR DESCRIPTION
**Pull request by cBournhonesque**
Fixes https://github.com/cBournhonesque/lightyear/issues/144

All credit for this PR goes to @Nul-led who came up with the idea and did the entire implementation. 
I just copy-pasted his code to put it in a `WebPlugin`. Thank you very much for this amazing solution!

In wasm, the main thread gets quickly throttled by the browser when it is hidden (e.g. when the user switches tabs).
This means that the app.update() function will not be called, because bevy's scheduler only runs `app.update()` when
the browser's requestAnimationFrame is called. (and that happens only when the tab is visible)

This is problematic because:
- we stop sending packets so the server disconnects the client because it doesn't receive keep-alives
- when the client reconnects, it also disconnects because it hasn't been receiving server packets
- the internal transport buffers can overflow because they are not being emptied

This solution spawns a WebWorker (a background thread) which is not throttled, and which runs
`app.update()` at a fixed interval. This way, the client can keep sending and receiving packets,
and updating the local World.

TODO:
- probably don't need `world.clear_trackers()`
- make interval configurable?
- confirm that the WebWorker doesn't get throttled even after long periods?

### CODE REVIEW

**Comment by simbleau** [comment]
This has universally great functionality - Should be a plugin one can easily `cargo add webworker_update` and add low-frequency updates to their application

File: lightyear/src/client/web.rs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    )
+    .unwrap();
+
+    let worker = Worker::new(&Url::create_object_url_with_blob(&blob).unwrap()).unwrap();
+
+    let closure = Closure::<dyn FnMut()>::new(move || {
+        if window().unwrap().document().unwrap().hidden() {
+            // Imitate app.update()
+            let world = unsafe { world_ptr.write().unwrap().as_mut().unwrap() };
+            world.run_schedule(Main);
```
Comment (simbleau) [inline lightyear/src/client/web.rs:46]: Magnifique

**Comment by Nul-led** [comment]
@simbleau thanks :p
ill probably make a crate for it then

**Comment by Nul-led** [comment]
re: configurable interval
1) we use format! to make it configurable at startup
2) we make an actual protocol for changing the duration (could be achieved by replacing setInterval with setTimeout)

**Comment by Nul-led** [comment]
re: trottled web workers
doesnt seem to happen (tested over 30 minutes), didnt confirm on webkit since i dont have a mac (feel free to check if anyone does), confirmed that it works for at least 7 minutes on android chromium based (more than that doesnt seem reasonable anyways)

**Comment by Nul-led** [comment]
Another note ill save here:
Lightyear doesnt need to send inputs while in background mode, so maybe worth looking into to save bandwidth?

**Comment by Nul-led** [comment]
> > @simbleau thanks :p ill probably make a crate for it then
> 
> Please let me know when it's done, I'd add this to my bevy_rtc project today.

maybe make an impl urself for now then, ill prob wont have time until weekend

**Pull request merged**